linux/SystemdMeter.c

```diff
@@ -83,8 +88,34 @@ static void SystemdMeter_done(ATTR_UNUSED Meter* this) {
 #endif /* BUILD_STATIC */
 }
 
+static void SystemdUserMeter_done(ATTR_UNUSED Meter* this) {
+   if (systemState) {
+      free(systemState);
+      systemState = NULL;
+   }
+
+#ifdef BUILD_STATIC
+# ifdef HAVE_LIBSYSTEMD
+   if (userbus) {
+      sym_sd_bus_unref(userbus);
+   }
+   userbus = NULL;
+# endif /* HAVE_LIBSYSTEMD */
+#else /* BUILD_STATIC */
+   if (userbus && dlopenHandle) {
+      sym_sd_bus_unref(userbus);
+   }
+   userbus = NULL;
+
+   if (dlopenHandle) {
+      dlclose(dlopenHandle);
+      dlopenHandle = NULL;
+   }
```

Comment:
This must not be freed unless neither `bus_system` nor `bus_user` are also already freed.

Reply:
I think this does what you mean